Sources/Private/CoreAnimation/CoreAnimationLayer.swift

```diff
@@ -343,15 +348,29 @@ extension CoreAnimationLayer: RootAnimationLayer {
   var isAnimationPlaying: Bool? {
     switch pendingAnimationConfiguration?.playbackState {
     case .playing:
-      true
+      return true
+
     case .paused:
-      false
+      return false
+
     case nil:
       switch playbackState {
       case .playing:
-        animation(forKey: #keyPath(animationProgress)) != nil
```

Comment:
This is clever, but changing it to no longer use the layer itself as the source of truth sounds like it will cause unexpected behavior changes. This is also the same approach used by the Main Thread rendering engine.

Is there another way to fix this issue without modifying this check? Perhaps we could update the specific call site of this check that is related to the bug you are investigating?

Reply:
I'm open to trying it though. I see how this is probably the only solution to the issue.